test/Sema/access-level-import-classic-exportability.swift

```diff
@@ -0,0 +1,207 @@
+extension PublicImportType {
+    public func publicMethod() {}
+}
+
+package protocol PublicConstrainedExtensionProtoInPackage {}
+extension Array: PublicConstrainedExtensionProtoInPackage where Element == PackageImportType {}
+extension PublicImportType {
+    package func packageMethod() {}
+}
+
+internal protocol PublicConstrainedExtensionProtoInInternal {}
+extension Array: PublicConstrainedExtensionProtoInInternal where Element == PublicImportType {}
+extension PublicImportType {
+    internal func internalMethod() {}
+}
+
+fileprivate protocol PublicConstrainedExtensionProtoInFileprivate {}
+extension Array: PublicConstrainedExtensionProtoInFileprivate where Element == PublicImportType {}
+extension PublicImportType {
+    fileprivate func fileprivateMethod() {}
+}
+
+private protocol PublicConstrainedExtensionProtoInPrivate {}
+extension Array: PublicConstrainedExtensionProtoInPrivate where Element == PublicImportType {}
+extension PublicImportType {
+    private func privateMethod() {}
+}
+
+public protocol PackageConstrainedExtensionProto {}
+extension Array: PackageConstrainedExtensionProto where Element == PackageImportType {} // expected-error {{cannot use struct 'PackageImportType' in an extension with conditional conformances; 'PackageLib' was not imported publicly}}
+extension PackageImportType { // expected-error {{cannot use struct 'PackageImportType' in an extension with public or '@usableFromInline' members; 'PackageLib' was not imported publicly}}
+    public func publicMethod() {}
+}
+
+package protocol PackageConstrainedExtensionProtoInPackage {}
+extension Array: PackageConstrainedExtensionProtoInPackage where Element == PackageImportType {}
+extension PackageImportType {
+    package func packageMethod() {}
+}
+
+internal protocol PackageConstrainedExtensionProtoInInternal {}
+extension Array: PackageConstrainedExtensionProtoInInternal where Element == PackageImportType {}
+extension PackageImportType {
+    internal func internalMethod() {}
+}
+
+fileprivate protocol PackageConstrainedExtensionProtoInFileprivate {}
+extension Array: PackageConstrainedExtensionProtoInFileprivate where Element == PackageImportType {}
+extension PackageImportType {
+    fileprivate func fileprivateMethod() {}
+}
+
+private protocol PackageConstrainedExtensionProtoInPrivate {}
+extension Array: PackageConstrainedExtensionProtoInPrivate where Element == PackageImportType {}
+extension PackageImportType {
+    private func privateMethod() {}
+}
+
+public protocol InternalConstrainedExtensionProto {}
+extension Array: InternalConstrainedExtensionProto where Element == InternalImportType {} // expected-error {{cannot use struct 'InternalImportType' in an extension with conditional conformances; 'InternalLib' was not imported publicly}}
+extension InternalImportType { // expected-error {{cannot use struct 'InternalImportType' in an extension with public or '@usableFromInline' members; 'InternalLib' was not imported publicly}}
```

Comment:
The access level of the extension currently overshadows its member's access level, so even if `foo` is `public`, `foo` automatically defaults to `internal` because of its enclosing extension's access level. Wouldn't this error conflict with the current behavior? Or is raising this current behavior as an error part of the goal?

Reply:
Yeah, we end up treating this use case differently for access-level on imports vs the classic local access level. I would be open to go the local way and not raising diagnostics and instead skipping these extensions at swiftinterface printing if this is causing problems. However I think if we were to do this again we'd pick this early error instead of the silent ignore logic we currently have at 3 different places in the compiler. This shouldn't affect the current behavior when not using access-levels on imports though.

Reply:
There are two levels here:
1. Non-public used in public is an error with this PR. I prefer to have a stronger check for a new feature as it won't break existing code. The error also follows the pattern of other disallowed origins used in API.
2. For anything under public, there's no diagnostics. We'll probably need one for package, then below that it may not matter. This should be an error, with likely a transition period. We could also add a warning for internal and below, but that may be superfluous, as long as there's not clear boundary like the one imposed by printing only public decls in the swiftinterface the mismatching access level may not matter much.